tslint.json

```diff
@@ -12,13 +12,13 @@
         "no-empty": [true, "allow-empty-catch", "allow-empty-functions"],
         "no-string-literal": false,
         "interface-over-type-literal": false,
-        // TODO: turn on
-        "prefer-const": false,
-        // I do want to bring this down to 120 at some point but maybe later
+        "prefer-const": {
+            "severity": "warning"
+        },
         "max-line-length": {
             "severity": "warning",
             "options": {
-                "limit": 140,
+                "limit": 120,
                 "ignore-pattern": "\\.replace\\(('\\[|/)|^\\s*(// \\s*)?(return |(let |const )?[a-zA-Z0-9$.]+ \\+?= (\\$\\()?|[a-zA-Z0-9$.]+\\()?['\"`]"
```

Comment:
```suggestion
                "ignore-pattern": "\\.replace\\(('\\[|/)|^\\s*(// \\s*)?(return |throw (new )?|(let |const )?[a-zA-Z0-9$.]+ \\+?= (\\$\\()?|[a-zA-Z0-9$.]+\\()?['\"`]"
```

Reply:
```suggestion
                "ignore-pattern": "^\\s*(// \\s*)?((let |const )?[a-zA-Z0-9$.]+ \\+?= (\\$\\()?|(return |throw )?(new )?([a-zA-Z0-9$.]+\\()?)?['\"`/]"
```

Reply:
Sorry for delaying until a conflict appeared; I didn't have much free time yesterday.

Reply:
Why remove `replace`? It means the zalgo regex at sim/dex.ts:253 now is too long?

Reply:
The zalgo regex should be:

```js
		name = name.replace(
			/[\u0300-\u036f\u0483-\u0489\u0610-\u0615\u064B-\u065F\u0670\u06D6-\u06DC\u06DF-\u06ED\u0E31\u0E34-\u0E3A\u0E47-\u0E4E]{3,}/g,
			''
		);
```